pkg/apis/pingcap.com/v1alpha1/types.go

```diff
@@ -258,3 +258,215 @@ type TiKVFailureStore struct {
+type BackupSchedule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   BackupScheduleSpec   `json:"spec"`
+	Status BackupScheduleStatus `json:"status,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// BackupScheduleList contains a list of BackupSchedule.
+type BackupScheduleList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Items []BackupSchedule `json:"items"`
+}
+
+// BackupScheduleSpec contains the backup schedule specification for a tidb cluster.
+type BackupScheduleSpec struct {
+	// Schedule specifies the cron string used for backup scheduling.
+	Schedule string `json:"schedule"`
+	// MaxBackups is to specify how many backups we want to keep.
+	MaxBackups int `json:"maxBackups"`
+	// BackupTemplate is the specification of the backup structure to get scheduled.
+	BackupTemplate BackupSpec `json:"backupTemplate"`
+}
+
+// BackupScheduleStatus represents the current state of a BackupSchedule.
+type BackupScheduleStatus struct {
+	// LastBackup represents the last backup.
+	LastBackup string `json:"lastBackup"`
+	// LastBackupTime represents the last time the backup was successfully created.
+	LastBackupTime metav1.Time `json:"lastBackupTime"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Restore represents the restoration of backup of a tidb cluster.
+type Restore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   RestoreSpec   `json:"spec"`
+	Status RestoreStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RestoreList contains a list of Restore.
+type RestoreList struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Items []Restore `json:"items"`
+}
+
+// RestoreConditionType represents a valid condition of a Restore.
+type RestoreConditionType string
+
+const (
+	// RestoreRunning means the Restore is currently being executed.
+	RestoreRunning RestoreConditionType = "Running"
+	// RestoreComplete means the Restore has successfully executed and the
+	// backup data has been loaded into tidb cluster.
+	RestoreComplete RestoreConditionType = "Complete"
+	// RestoreFailed means the Restore has failed.
+	RestoreFailed RestoreConditionType = "Failed"
+)
+
+// RestoreCondition describes the observed state of a Restore at a certain point.
+type RestoreCondition struct {
+	Type    RestoreConditionType
+	Status  corev1.ConditionStatus
+	Reason  string
+	Message string
+}
+
+// RestoreSpec contains the specification for a restore of a tidb cluster backup.
+type RestoreSpec struct {
+	// Cluster represents the tidb cluster to be restored.
+	Cluster string `json:"cluster"`
+	// Backup represents the backup object to be restored.
+	Backup string `json:"backup"`
+	// SecretName is the name of the secret which stores
```

Comment:
`Cluster` and `Backup` are string type, what information does they store?

Reply:
Point to TidbCluster CR and Backup CR name